sdk/servicebus/azure-messaging-servicebus/src/samples/java/com/azure/messaging/servicebus/DeadletterQueueSample.java

```diff
@@ -0,0 +1,222 @@
+                JSONObject jsonObject = null;
+                try {
+                    jsonObject = JSONObjectUtils.parse(new String(body, UTF_8));
+                } catch (ParseException e) {
+                    e.printStackTrace();
+                }
+                System.out.printf(
+                    "\n\t\t\t\tMessage received: \n\t\t\t\t\t\tMessageId = %s, \n\t\t\t\t\t\tSequenceNumber = %s, \n\t\t\t\t\t\tEnqueuedTimeUtc = %s,"
+                        + "\n\t\t\t\t\t\tExpiresAtUtc = %s, \n\t\t\t\t\t\tContentType = \"%s\",  \n\t\t\t\t\t\tContent: [ firstName = %s, name = %s ]\n",
+                    receiveMessage.getMessageId(),
+                    receiveMessage.getSequenceNumber(),
+                    receiveMessage.getEnqueuedTime(),
+                    receiveMessage.getExpiresAt(),
+                    receiveMessage.getContentType(),
+                    jsonObject != null ? jsonObject.get("firstName") : "",
+                    jsonObject != null ? jsonObject.get("name") : "");
+            } else {
+                receiverAsyncClient.deadLetter(receiveMessage);
+            }
+            receiverAsyncClient.complete(receiveMessage);
+        });
+    }
+
+    /**
+     * Receive dead letter {@link ServiceBusMessage messages} and resend its.
+     *
+     * @Param connectionString Service Bus Connection String
+     * @Param queueName Queue Name
+     * @Param resubmitSender Service Bus Send Client
+     */
+    void pickUpAndFixDeadletters(String connectionString, String queueName, ServiceBusSenderClient resubmitSender) {
+        ServiceBusReceiverAsyncClient receiverAsyncClient
+            = new ServiceBusClientBuilder()
+            .connectionString(connectionString)
+            .receiver()
+            .receiveMode(ServiceBusReceiveMode.PEEK_LOCK)
+            .queueName(queueName.concat("/$deadletterqueue"))
+            .buildAsyncClient();
+
+        receiverAsyncClient.receiveMessages().subscribe(receiveMessage -> {
+            if (receiveMessage.getSubject() != null && receiveMessage.getSubject().contentEquals("Physicist")) {
+                ServiceBusMessage resubmitMessage = new ServiceBusMessage(receiveMessage.getBody());
+                System.out.printf(
+                    "\n\t\tFixing: \n\t\t\tMessageId = %s, \n\t\t\tSequenceNumber = %s, \n\t\t\tLabel = %s\n",
+                    receiveMessage.getMessageId(),
+                    receiveMessage.getSequenceNumber(),
+                    receiveMessage.getSubject());
+                resubmitMessage.setMessageId(receiveMessage.getMessageId());
+                resubmitMessage.setSubject("Scientist");
+                resubmitMessage.setContentType(receiveMessage.getContentType());
+                resubmitMessage.setTimeToLive(Duration.ofMinutes(2));
+                resubmitSender.sendMessage(resubmitMessage);
+            }
+            receiverAsyncClient.complete(receiveMessage);
```

Comment:
Same comment about Mono<Void>. Also, this operation is never subscribed to, so it will never be invoked. I would chain this into the stream by using flatMap.

Reply:
According to your Comment, fixed in new version.